### ISSUE TITLE
[Feature Request] A Drawer Component for NextUI

### ISSUE THREAD

**Issue by moeen-mahmud** [nextui-org/nextui#813]
### Is your feature request related to a problem? Please describe.

I've been using NextUI in one of my side projects lately. I appreciate the simple and robust approach it offers, and I adore the UI overall. However, I am surprised it doesn't have a **Drawer** component! Although, I implemented it using a third-party library. But it's not enough. Third-party components lack native NextUI effects and support. It's also not the best approach, I guess. 

NextUI should also have a **Drawer** like other UI libraries, e.g., Material UI, Ant Design, Mantine, etc.

### Describe the solution you'd like

It'd be great to have a native **Drawer** component for NextUI. It'll save a lot of tedious jobs and broken tests. 

As the **Hacktoberfest** has begun, it brings a fantastic opportunity to address this lacking feature and implement it right now. 

I'd like to suggest that you guys should open an issue for this to implement that. I was also tinkering with the NextUI repo myself. 

I'd love to work on that feature and contribute. 

### Describe alternatives you've considered

I'm temporarily using [React Modern Drawer](https://github.com/Farzin-Firoozi/react-modern-drawer). It's minimal and easy to customize.

**Comment by ytkg** [comment]
+1

**Comment by gajendr4** [comment]
A temporary alternative would be to use modal component with full-screen prop with inline css and media queries to setup max width to something like 30vw-70vw on different devices

**Comment by moeen-mahmud** [comment]
> A temporary alternative would be to use modal component with full-screen prop with inline css and media queries to setup max width to something like 30vw-70vw on different devices

@gajendr4 Yeah, that's a straightforward option. But still, it would be great if we had such a component.

**Comment by maxtondelaney** [comment]
If anyones looking for an implementation of a drawer using the modal component, here's something I put together as a temporary solution that fakes it until a drawer is fully implemented - may be a good starting point if anyone needs it.

`Drawer.tsx`
```
import React from 'react';
import { Modal, ModalProps, useModal } from '@nextui-org/react';

import './styles.css';

type SubComponents = {
  Header: typeof Modal.Header;
  Body: typeof Modal.Body;
  Footer: typeof Modal.Footer;
};

type Props = Omit<
  ModalProps,
  'className' | 'fullScreen' | 'closeButton' | 'animated' | 'blur'
>;

const Drawer: React.FC<Props> & SubComponents = ({ children, ...props }) => {
  const { open } = props;

  return (
    <Modal
      className={`drawer drawer-animated ${
        open ? 'drawer-animated-slide-in' : ''
      }`}      
      closeButton
      animated={false}
      {...props}
    >
      {children}
    </Modal>
  );
};

Drawer.Header = Modal.Header;
Drawer.Body = Modal.Body;
Drawer.Footer = Modal.Body;

export const useDrawer = useModal;

export default Drawer;
```

`styles.css`
```
.drawer {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  max-width: 650px;
  border-radius: 0;
}

.drawer-animated {
  transform: translateX(-100%);
}

.drawer-animated-slide-in {
  animation: slide-in 0.3s forwards;
}

@keyframes slide-in {
  from {
    transform: translateX(-100%);
  }
  to {
    transform: translateX(0%);
  }
}
```

**Comment by moeen-mahmud** [comment]
Hey @maxtondelaney! 

That looks cool and robust since we don't have any drawer components! Looking forward to explore more.

Thanks for the example
